tools/fix_auth/fix_auth.rb

```diff
@@ -84,17 +84,20 @@ def load_rails
 
     def set_passwords
       ManageIQ::Password.key_root = cert_dir if cert_dir
-      if options[:legacy_key]
-        if (legacy_key = ManageIQ::Password.load_key_file(options[:legacy_key]))
-          ManageIQ::Password.keys["alt"] = legacy_key
-        else
-          puts "WARNING: key #{options[:legacy_key]} not found"
-        end
+    end
+
+    def set_legacy_key
+      key = ManageIQ::Password.load_key_file(options[:legacy_key])
+      if key
+        options[:legacy_key] = key
+      else
+        puts "WARNING: legacy key #{options[:legacy_key]} not found"
```

Comment:
do you want to clear out `options[:legacy_key]`? think it is now expecting a MiqPassword record rather than the legacy key filename

or possibly that is intentionally setup this way for tests?

Reply:
It's still expecting a file name as the parameter to fixauth, and then on line 92 we swap it for an actual MiqPassword object created from that file.  This line is when they pass an invalid key filename, and I need to notify the user that the filename is incorrect.

Reply:
ohhhh I see what you are saying..you mean set it to nil, so later code doesn't try to "recrypt" but with a file name.  Great catch...I'll do that.

Reply:
Updated